internal/advtools/convo.lua

```diff
@@ -61,8 +61,111 @@ local function generateKeywordsForChoice(choice)
     addKeywords(choice, new_keywords)
 end
 
+-- Helper function to create new dialog choices, returns the created choice
+local function new_choice(choice_type, title, keywords)
+    local choice = df.adventure_conversation_choice_infost:new()
+    choice.cc = df.talk_choice:new()
+    choice.cc.type = choice_type
+    local text = df.new("string")
+    text.value = title
+    choice.print_string.text:insert("#", text)
+
+    if keywords ~= nil then
+        addKeywords(choice, keywords)
+    end
+    return choice
+end
+
+local function addHistFigWhereaboutsChoice(profile)
+    for i, c in pairs(adventure.conversation.conv_choice_info) do
+        if c.cc.type == df.talk_choice_type.AskWhereabouts and
+          c.cc.invocation_target_hfid ~= -1 and
```

Comment:
if `profile.histfig_id` is never equal to -1, then  you can remove the check for -1